components/constellation/constellation.rs

```diff
@@ -1153,14 +1179,22 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         // the new load. The framing element must be notified about the
         // requested change so it can update its internal state.
         let parent_info = self.pipelines.get(&source_id).and_then(|source| source.parent_info);
+
+        // Message for the script thread, so that it can send activation event to
+        // service worker managers, if any registration exists in the containing script thread.
+        let msg_for_sw_manager = ConstellationControlMsg::NavigateForServiceWorker(load_data.clone(),
+                                                          self.serviceworker_manager.clone().unwrap());
         match parent_info {
             Some((parent_pipeline_id, subpage_id, _)) => {
                 self.handle_load_start_msg(&source_id);
                 // Message the constellation to find the script thread for this iframe
                 // and issue an iframe load through there.
                 let msg = ConstellationControlMsg::Navigate(parent_pipeline_id, subpage_id, load_data);
                 let result = match self.pipelines.get(&parent_pipeline_id) {
-                    Some(parent_pipeline) => parent_pipeline.script_chan.send(msg),
+                    Some(parent_pipeline) => {
+                        parent_pipeline.script_chan.send(msg_for_sw_manager).unwrap();
```

Comment:
@jdm The message of a load url, is needed, to send to script thread, so that it can matches the stored registrations scope url, and trigger the manager of activation of any service worker.